src/cdk/explorer/nodes/constructNode.ts

```diff
@@ -0,0 +1,77 @@
+/*!
+ * Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { AWSTreeNodeBase } from '../../../shared/treeview/nodes/awsTreeNodeBase'
+import { cdk } from '../../globals'
+import { CfnResourceKeys, ConstructTreeEntity } from '../tree/types'
+
+/**
+ * Represents a CDK construct
+ */
+export class ConstructNode extends AWSTreeNodeBase {
+    private readonly type: string
+
+    get tooltip(): string {
+        return this.type || this.construct.path
+    }
+
+    get id(): string {
+        return this.construct.path
+    }
+
+    get iconPath(): string | vscode.Uri | { light: string | vscode.Uri; dark: string | vscode.Uri } | vscode.ThemeIcon {
```

Comment:
Gotta love them combined types in the VS Code API. Because the icon is a function of type, and type is fixed for the lifetime of the node, you can simply set `this.iconPath` from the constructor (similar to `cloudformationodes.ts`).

As a bonus, I think you can move the type-to-icon logic for mapping into a global/utility method outside this class. It would be nice if the icon logic could be more maintainable than a series of if statements in the future.